README.md

```diff
@@ -3,16 +3,112 @@ Dagger Reflect
 
 A reflection-based implementation of the [Dagger][dagger] dependency injection library for fast IDE builds.
 
-More info soon...
+See [my talk (from about 37:25)][talk] for more details on how this came about.
+
+
+Er, what? Why would I want this?
+--------------------------------
+
+Feels like going back to Dagger 1 by Square? Not quite: we keep the benefits of Dagger 2, including the annotation processing compile-time validation in production, but using reflection speeds up development builds.
+
+The normal `dagger` artifact requires the use of `dagger-compiler` as an annotation processor for compile-time validation of your components and code generation for runtime performance.
+This is a desirable feature for your CI and release builds, but it slows down iterative development.
+By using `dagger-reflect` for only your IDE builds, you have one less annotation processor sitting between you and your running app. This is especially important for Kotlin-only or Java/Kotlin mixed projects using KAPT. And if `dagger-compiler` is your only annotation
+processor for a module, using `dagger-reflect` means that **zero** annotation processors run during development.
+
+
+Can I use this in production?
+-----------------------------
+
+No.
+
+Well technically you _can_, but don't. It's slow, inefficient, and lacks the level of validation that normal Dagger usage provides.
 
 
 Usage
 -----
 
-Details soon...
+_Replace (or fulfill via gradle.properties) `${VERSION_DAGGER}` and `${VERSION_DAGGER_REFLECT}` with whatever versions you are using._
+
+If you have a dedicated variant for development you can skip the `if` check and add the corresponding dependencies for the development and non-development variants.
+
+### Easy migration, quick APT
+This usage type is compatible with the current dagger usage patterns where users of Dagger call into the generated code via `DaggerFooComponent.create()` or `DaggerFooComponent.builder()`. The `reflect-compiler` generates source compatible code as `dagger-compiler` would, which resolves the components via reflection. The drawback is that there's still an annotation processor present, albeit much faster than Dagger.
+
+```gradle
+dependencies {
+  implementation "com.google.dagger:dagger:${VERSION_DAGGER}"
+  if (project.hasProperty('android.injected.invoked.from.ide')) {
+    implementation "com.jakewharton.dagger:dagger-reflect:${VERSION_DAGGER_REFLECT}"
+    annotationProcessor "com.jakewharton.dagger:dagger-reflect-compiler:${VERSION_DAGGER_REFLECT}"
+  } else {
+    annotationProcessor "com.google.dagger:dagger-compiler:${VERSION_DAGGER}"
+  }
+}
+```
+
+_(No need to upgrade Proguard shirking configuration with this approach.)_
```

Comment:
```suggestion
_(No need to upgrade Proguard shrinking configuration with this approach.)_
```

Reply:
weird, I actually remember fixing that typo... maybe it was somewhere else.